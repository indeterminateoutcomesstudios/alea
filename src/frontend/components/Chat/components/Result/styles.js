// @flow
import styled from 'styled-components'
import { colors } from 'frontend/styles/common'

export const Container = styled.div`
  margin-top: 0.6rem;
  padding: 0;
  background: ${props => props.theme.backgroundSecondary};
  border-left: 3px solid ${colors.blue};
  padding-left: 5px;
  border-radius: 5px;
`

export const Row = styled.div`
  border-top: 1px solid ${props => props.theme.borderColor};

  &:first-child {
    border-top: 0;
  }
`

export const Modifier = styled.span`
  font-weight: 400;
  color: ${props => props.theme.colorSecondary};
  margin-right: 0.4em;
`

export const Roll = styled.span`
  font-weight: 600;
  margin-right: 0.4em;
`
