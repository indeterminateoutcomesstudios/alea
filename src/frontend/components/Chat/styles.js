// @flow
import styled from 'styled-components'

export const Container = styled.div`
  background-color: ${props => props.theme.background};
  width: 320px;
  overflow: hidden;
  display: flex;
  flex-direction: column;

  ${props =>
    props.isPinned
      ? `
    position: absolute;
    top: 4rem;
    right: 0;
    bottom: 0;
  `
      : `
    position: absolute;
    top: auto;
    right: 1rem;
    bottom: 1rem;
    box-shadow: 0 3px 5px 0 rgba(0, 0, 0, 0.2);
    border-radius: 5px;
  `};
`

export const Loading = styled.div`
  padding: 10px;
`

export const Error = Loading

export const MessagesWrapper = styled.div`
  min-height: 0;
  flex: 1;
`
export const Messages = styled.div`
  overflow-y: ${props => (props.isPinned ? 'scroll' : 'hidden')};
  overflow-x: hidden;
  width: 315px;
  height: 100%;

  display: flex;
  flex-direction: column;
  flex: ${props => (props.isPinned ? '1' : 'none')};
`
