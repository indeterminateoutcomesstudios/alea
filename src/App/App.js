/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import { Switch, Route } from 'react-router'
import styled, { ThemeProvider } from 'styled-components'
import Loading from 'components/Loading'
import Login from 'containers/Login'
import Game from './Game'
import Header from '../containers/Header'
import Sessions from './Sessions'
import Settings from '../containers/Settings'
import * as themes from 'styles/themes'
import type { State } from 'store'

type Props = {
  appIsLoading: boolean,
  userIsLoggedIn: boolean,
  showSettings: boolean,
  location: Object,
  theme: Object
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: ${props => props.theme.map};
`

class App extends React.Component<Props> {
  render() {
    const { appIsLoading, userIsLoggedIn, showSettings, theme } = this.props

    if (appIsLoading) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Loading />
        </ThemeProvider>
      )
    }

    if (!userIsLoggedIn) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Login />
        </ThemeProvider>
      )
    }

    const settings = showSettings ? <Settings /> : null

    return (
      <ThemeProvider theme={theme}>
        <Container>
          <Header />
          <Switch>
            <Route exact path="/" component={Sessions} />
            <Route path="/g/:id/" component={Game} />
          </Switch>
          {settings}
        </Container>
      </ThemeProvider>
    )
  }
}

const mapStateToProps = (state: State): Props => ({
  appIsLoading: state.ui.appIsLoading,
  userIsLoggedIn: state.ui.userIsLoggedIn,
  theme: themes[state.user.preferences.theme],
  showSettings: state.ui.showSettings,
  location: state.router.location
})

export default connect(mapStateToProps)(App)
