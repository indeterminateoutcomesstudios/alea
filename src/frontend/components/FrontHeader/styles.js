// @flow
import { Link } from 'react-router-dom'
import styled from 'styled-components'
import { fonts } from 'frontend/styles/common'

export const Container = styled.div`
  display: flex;
  flex-direction: row;
  color: white;
  height: 8rem;
`

export const LogoButton = styled(Link)`
  padding-top: 27px;
  padding-left: 2rem;
  text-decoration: none;
  color: white;
`

export const Button = styled(Link)`
  background-color: white;
  border-radius: 5px;
  height: 4rem;
  line-height: 4rem;
  padding: 0 2rem;
  margin: 2rem;
  margin-left: auto;
  color: ${props => props.theme.backgroundInverted};
  font-family: ${fonts.heading};
  font-weight: 700;
  text-decoration: none;
`
