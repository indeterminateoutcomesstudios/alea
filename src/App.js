/* @flow */
import React from 'react'
import { connect } from 'react-redux'
import { Switch, Route } from 'react-router'
import styled, { ThemeProvider } from 'styled-components'
import Loading from './components/Loading'
import Login from './containers/Login'
import Header from './containers/Header'
import Sessions from './containers/Sessions'
import Settings from './containers/Settings'
import Chat from './containers/Chat'
import Map from './containers/Map'
import Sidebar from './containers/Sidebar'
import * as themes from 'styles/themes'
import { CONSTS } from 'styles/common'
import type { State } from 'store'

type Props = {
  appIsLoading: boolean,
  userIsLoggedIn: boolean,
  showSettings: boolean,
  location: Object,
  theme: Object
}

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  background-color: ${props => props.theme.map};
`

const GameInner = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  flex: 1;
  position: absolute;
  top: ${CONSTS.headerHeight};
  right: 0;
  bottom: 0;
  left: 0;
`

const Game = () => (
  <GameInner>
    <Sidebar />
    <Map />
    <Chat />
  </GameInner>
)

class App extends React.Component<Props> {
  render() {
    const { appIsLoading, userIsLoggedIn, showSettings, theme } = this.props

    if (appIsLoading) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Loading />
        </ThemeProvider>
      )
    }

    if (!userIsLoggedIn) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Login />
        </ThemeProvider>
      )
    }

    const settings = showSettings ? <Settings /> : null

    return (
      <ThemeProvider theme={theme}>
        <Container>
          <Header />
          <Switch>
            <Route exact path="/" component={Sessions} />
            <Route path="/g/:id/" component={Game} />
          </Switch>
          {settings}
        </Container>
      </ThemeProvider>
    )
  }
}

const mapStateToProps = (state: State): Props => ({
  appIsLoading: state.ui.appIsLoading,
  userIsLoggedIn: state.ui.userIsLoggedIn,
  theme: themes[state.user.preferences.theme],
  showSettings: state.ui.showSettings,
  location: state.router.location
})

export default connect(mapStateToProps)(App)
