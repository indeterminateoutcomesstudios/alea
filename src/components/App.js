// @flow
import React from 'react'
import { Route, Switch } from 'react-router'
import styled, { ThemeProvider } from 'styled-components'
import Login from '../containers/Login'
import Header from '../containers/Header'
import Sessions from '../containers/Sessions'
import Settings from '../containers/Settings'
import Chat from '../containers/Chat'
import Map from '../containers/Map'
import Sidebar from '../containers/Sidebar'
import * as themes from '../styles/themes'
import { CONSTS } from '../styles/common'
import Loading from './Loading'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  background-color: ${props => props.theme.map};
`

const GameInner = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  flex: 1;
  position: absolute;
  top: ${CONSTS.headerHeight};
  right: 0;
  bottom: 0;
  left: 0;
`

const Game = () => (
  <GameInner>
    <Sidebar />
    <Map />
    <Chat />
  </GameInner>
)

export type Props = {
  appIsLoading: boolean,
  userIsLoggedIn: boolean,
  location: Object,
  theme: Object
}
class App extends React.Component<Props> {
  render() {
    const { appIsLoading, userIsLoggedIn, theme } = this.props

    if (appIsLoading) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Loading />
        </ThemeProvider>
      )
    }

    if (!userIsLoggedIn) {
      return (
        <ThemeProvider theme={themes['light']}>
          <Login />
        </ThemeProvider>
      )
    }

    return (
      <ThemeProvider theme={theme}>
        <Container>
          <Header />
          <Switch>
            <Route exact path="/" component={Sessions} />
            <Route path="/g/:name/:id" component={Game} />
          </Switch>
          <Settings />
        </Container>
      </ThemeProvider>
    )
  }
}

export default App
