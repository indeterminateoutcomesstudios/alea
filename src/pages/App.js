// @flow
import React from 'react'
import { Route, Switch } from 'react-router'
import { connect } from 'react-redux'
import styled from 'styled-components'
import type { State } from '../store'
import Sessions from '../containers/Sessions'
import Chat from '../containers/Chat'
import Map from '../containers/Map'
import Sidebar from '../containers/Sidebar'
import { CONSTS } from '../styles/common'
import Loading from '../components/Loading'

const Container = styled.div`
  display: flex;
  flex-direction: column;
  align-items: stretch;
  background-color: ${props => props.theme.map};
`

const GameInner = styled.div`
  display: flex;
  flex-direction: row;
  align-items: stretch;
  flex: 1;
  position: absolute;
  top: ${CONSTS.headerHeight};
  right: 0;
  bottom: 0;
  left: 0;
`

const Game = () => (
  <GameInner>
    <Sidebar />
    <Map />
    <Chat />
  </GameInner>
)

type Props = {
  appIsLoading: boolean,
  location: Object
}
export class App extends React.Component<Props> {
  render() {
    const { appIsLoading } = this.props

    if (appIsLoading) {
      return <Loading />
    }

    return (
      <Container>
        <Switch>
          <Route exact path="/sessions" component={Sessions} />
          <Route path="/g/:name/:id" component={Game} />
        </Switch>
      </Container>
    )
  }
}

const mapStateToProps = (state: State): Props => ({
  appIsLoading: state.ui.appIsLoading,
  location: state.router.location
})

export default connect(mapStateToProps)(App)
