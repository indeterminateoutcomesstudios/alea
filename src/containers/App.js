// @flow
import { hot } from 'react-hot-loader'
import { connect } from 'react-redux'
import App from '../components/App'
import * as themes from 'styles/themes'
import type { State } from 'store'
import type { Props } from '../components/App'

const mapStateToProps = (state: State): Props => ({
  appIsLoading: state.ui.appIsLoading,
  userIsLoggedIn: state.ui.userIsLoggedIn,
  theme: themes[state.preferences.theme],
  location: state.router.location
})

const Container = connect(mapStateToProps)(App)

export default hot(module)(Container)
