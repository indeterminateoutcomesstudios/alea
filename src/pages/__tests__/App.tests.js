// @flow
import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { App } from '../App'
import { light } from '../../styles/themes'

describe('App component', () => {
  const renderer = new ShallowRenderer()

  it('renders correctly when loading', () => {
    renderer.render(
      <App appIsLoading showSettings={false} location={{}} theme={light} />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when finished loading', () => {
    renderer.render(
      <App
        appIsLoading={false}
        showSettings={false}
        location={{}}
        theme={light}
      />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when settings should be showing', () => {
    renderer.render(
      <App appIsLoading={false} showSettings location={{}} theme={light} />
    )
    const tree = renderer.getRenderOutput()
    expect(tree).toMatchSnapshot()
  })
})
