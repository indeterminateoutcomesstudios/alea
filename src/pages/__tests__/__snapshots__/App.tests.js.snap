// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly when finished loading 1`] = `
<styled.div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/sessions"
    />
    <Route
      component={[Function]}
      path="/g/:name/:id"
    />
  </Switch>
</styled.div>
`;

exports[`App component renders correctly when loading 1`] = `<LoadingModal />`;

exports[`App component renders correctly when settings should be showing 1`] = `
<styled.div>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/sessions"
    />
    <Route
      component={[Function]}
      path="/g/:name/:id"
    />
  </Switch>
</styled.div>
`;
